scripts/apps/authoring/authoring/services/AuthoringWorkspaceService.ts

```diff
@@ -158,6 +159,59 @@ export function AuthoringWorkspaceService($location, superdeskFlags, authoring,
         self.edit(item, 'correct');
     };
 
+    this.unpublish = (item) => {
+        $modal.open({
+            template: `<sd-unpublish-confirm-modal
+                data-item="item"
+                data-related="relatedItems"
+                data-close="close"
+                data-unpublish="unpublish">
+            '</sd-unpublish-confirm-modal>`,
+            controller: ['$scope', '$modalInstance', 'familyService',
+                ($scope, $modalInstance, familyService) => {
+                    $scope.item = item;
+                    $scope.relatedItems = [];
+
+                    familyService.fetchItems(item.archive_item.family_id, item)
+                        .then((items) => {
+                            const published = {};
+
+                            $scope.relatedItems = items._items
+                                .filter((relatedItem) => {
+                                    if (!published[relatedItem.guid]) {
+                                        published[relatedItem.guid] = true; // only show each item once
```

Comment:
Why does it return each item more than once?

Reply:
if you publish something, then unpublish, and publish again you can get same item there multiple times

Reply:
Which one of those multiple items do you want, last updated one?

> if you publish something, then unpublish, and publish again you can get same item there multiple times

This seems buggy. Why does it behave this way? Shouldn't publishing be only a matter of setting the state to `published` and unpublishing setting it back to `in-progress`?

Reply:
sort of, but for the record we store in published everything that was sent out - even when something is unpublished/killed there is a copy with different metadata sent to subscribers